src/oidc_child/oidc_child_json.c

```diff
@@ -413,6 +413,12 @@ errno_t parse_token_result(struct devicecode_ctx *dc_ctx,
         if (strcmp(json_string_value(tmp), "authorization_pending") == 0) {
             json_decref(result);
             return EAGAIN;
+        } else if (strcmp(json_string_value(tmp), "slow_down") == 0) {
+            /* RFC 8628: "... the interval MUST be increased by 5 seconds for"
+             *           "this and all subsequent requests." */
+            dc_ctx->interval += 5;
+            json_decref(result);
+            return EAGAIN;
```

Comment:
What if this error code is returned multiple times? Shouldn't there be some upper limit to the interval?

Reply:
Hi,

thanks, good catch. I didn't add a check here but made sure in the caller that the loop is not running longer than needed.

bye,
Sumit